litestar/middleware/base.py

```diff
@@ -149,3 +155,83 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             None
         """
         raise NotImplementedError("abstract method must be implemented")
+
+
+class ASGIMiddleware(abc.ABC):
+    """An abstract base class to easily construct ASGI middlewares, providing functionality
+    to dynamically skip the middleware based on ASGI ``scope["type"]``, handler ``opt``
+    keys or path patterns and a simple way to pass configuration to middlewares.
+
+    This base class does not implement an ``__init__`` method, so subclasses are free
+    to use it to customize the middleware's configuration.
+
+    .. important::
+        An instance of the individual middleware's will be created *once* and used to
+        build up the internal middleware stack. As such, middlewares should *not* be
+        stateful, as this state will be shared across all requests.
```

Comment:
```suggestion
        stateful, as this state will be shared across all requests.
        Any connection-specific state should be scoped to the `handle` implementation. Not doing so would typically lead to conflicting variable reads/writes across requests, and most likely bugs.
```

Reply:
Adapted from the Starlette docs here: https://www.starlette.io/middleware/#asgi-middleware-should-be-stateless 
The do/dont could be soething interesting to add to our docs too

Reply:
> The do/dont could be soething interesting to add to our docs too

Yeah, good idea